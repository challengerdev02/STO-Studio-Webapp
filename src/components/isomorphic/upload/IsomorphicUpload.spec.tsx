import { cleanup, queryByRole, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { IsomorphicUpload } from './index';
import React from 'react';

describe('IsomorphicUpload component', function () {
  afterEach(cleanup);
  let props = {
    children: <div>Upload file</div>,
    draggerProps: {
      action: 'https://test.com',
    },
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<IsomorphicUpload {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should Render Upload Component', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByRole(document.body, 'button')).toBeInTheDocument();
  });

  it('should Render Upload Component with Children', () => {
    // @ts-ignore
    setup({ children: <p>Hello</p> });
    expect(queryByRole(document.body, 'button')).toBeInTheDocument();
  });

  it('should Render Upload Component with No Children', () => {
    // @ts-ignore
    setup({ children: false });
    expect(queryByRole(document.body, 'button')).toBeInTheDocument();
  });

  it('should Render Upload Component with Crop', () => {
    // @ts-ignore
    const { container } = setup({ allowCrop: true });
    expect(queryByRole(document.body, 'button')).toBeInTheDocument();
  });
});
