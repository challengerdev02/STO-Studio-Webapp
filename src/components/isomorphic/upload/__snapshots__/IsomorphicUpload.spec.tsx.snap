// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsomorphicUpload component should match with snapshot 1`] = `
<div>
  <div
    class="ant-row ant-form-item sc-hKwDye cIllGI"
  >
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <span>
            <div
              class="ant-upload ant-upload-drag sc-bdvvtL cpVgWO"
            >
              <span
                class="ant-upload ant-upload-btn"
                role="button"
                tabindex="0"
              >
                <input
                  accept=""
                  style="display: none;"
                  type="file"
                />
                <div
                  class="ant-upload-drag-container"
                >
                  <div>
                    Upload file
                  </div>
                </div>
              </span>
            </div>
            <div
              class="ant-upload-list ant-upload-list-text"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
